packages/ui/src/components/Visualization/Canvas/Form/fields/ExpressionField/expression.service.ts

```diff
@@ -115,4 +115,26 @@ export class ExpressionService {
       return { [langName]: { expression: lang } } as ExpressionDefinition;
     }
   }
+
+  static updateExpressionFromModel(
+    sourceModel: Record<string, unknown> | undefined,
+    targetModel: Record<string, unknown>,
+  ): void {
+    if (!isDefined(sourceModel) || !isDefined(targetModel)) {
+      return;
+    }
+    const languageNames = this.getLanguageNames();
+    const sourceKey = Object.keys(sourceModel).find((key) => languageNames.includes(key));
+    const sourceExpressionString = sourceKey
+      ? (sourceModel[sourceKey] as Record<string, unknown>)?.expression
+      : undefined;
+
+    if (typeof sourceExpressionString === 'string') {
+      const targetKey = Object.keys(targetModel).find((key) => languageNames.includes(key));
+
+      if (targetKey) {
```

Comment:
[Suggestion] Should we also verify whether the selected language includes the expression key? Some languages, like Bean Method, don’t have it, and in such cases, we end up explicitly adding the expression value unless we perform this check.

Reply:
Might be even better to unit test what Shivam is talking about